# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.cpp
# Commit: 4fdcaa5d6ec3
# Full Hash: 4fdcaa5d6ec3368c91b8639bcc73b29425fc05ec
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-06-15 16:30:35
# Description:
#   Bug 1642919 - Remove MOZ_ASSERT in onAllowAccessFor r=timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79648
# ==============================================================================

diff -r e459bd0a9c55 -r 4fdcaa5d6ec3 toolkit/components/antitracking/ContentBlocking.cpp
--- a/toolkit/components/antitracking/ContentBlocking.cpp	Fri Jun 12 16:43:19 2020 +0000
+++ b/toolkit/components/antitracking/ContentBlocking.cpp	Mon Jun 15 08:07:16 2020 +0000
@@ -578,7 +578,9 @@
   // Let's inform the parent window.
   nsCOMPtr<nsPIDOMWindowInner> parentInner =
       AntiTrackingUtils::GetInnerWindow(aParentContext);
-  MOZ_ASSERT(parentInner);
+  if (NS_WARN_IF(!parentInner)) {
+    return;
+  }
 
   Document* doc = parentInner->GetExtantDoc();
   if (NS_WARN_IF(!doc)) {
